frame/bonded-finance/src/lib.rs

```diff
@@ -115,10 +117,11 @@ pub mod pallet {
 		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
 		/// The native currency, used for the stake required to create an offer.
-		type NativeCurrency: FungibleTransfer<AccountIdOf<Self>>;
+		type NativeCurrency: fungible::Mutate<AccountIdOf<Self>>
+			+ fungible::Transfer<AccountIdOf<Self>>;
 
 		/// The multi currency system offers are based on.
-		type Currency: FungiblesTransfer<AccountIdOf<Self>>;
+		type Currency: fungibles::Mutate<AccountIdOf<Self>> + FungiblesTransfer<AccountIdOf<Self>>;
```

Comment:
There is NO way to mint generic tokens without these bounds. IMO, it is not a problem because the runtime implementation already implements them

Reply:
AFAIK the pallet is not minting any token as both the bond asset and the reward are provided by users interacting with it. The reward is provided by the bond offer issuer and the bond asset is provided by users willing to bond?

Reply:
As well as unit tests, benchmarks also need initial minting in order to test internal logic. The different between both is that unit tests interacts with concrete types while benchmarks have to figure out abstract programming in order to measure different `runtime` performances, thus the reason of `fungibles::Mutate` along side the already existing `fungibles::Inspect` (alias of `FungiblesTransfer`).

Reply:
Alright then, I thought it was about the pallet features. We could probably have the tokens in the genesis block using the ExtBuilder but it's not a big deal, thanks for clarifying :).